api/v1alpha1/edgedevice_types.go

```diff
@@ -64,10 +64,25 @@ type HardwareProfileConfiguration struct {
 
 // EdgeDeviceStatus defines the observed state of EdgeDevice
 type EdgeDeviceStatus struct {
-	Phase                     string      `json:"phase,omitempty"`
-	LastSeenTime              metav1.Time `json:"lastSeenTime,omitempty"`
-	LastSyncedResourceVersion string      `json:"lastSyncedResourceVersion,omitempty"`
-	Hardware                  *Hardware   `json:"hardware,omitempty"`
+	Phase                     string       `json:"phase,omitempty"`
+	LastSeenTime              metav1.Time  `json:"lastSeenTime,omitempty"`
+	LastSyncedResourceVersion string       `json:"lastSyncedResourceVersion,omitempty"`
+	Hardware                  *Hardware    `json:"hardware,omitempty"`
+	Deployments               []Deployment `json:"deployments,omitempty"`
+}
+
+type EdgeDeploymentPhase string
+
+const (
+	Deploying EdgeDeploymentPhase = "Deploying"
```

Comment:
none of those reflects an erroneous status, how should admin indicate if the EdgeDeployment failed to run ?

Reply:
The data model accepts any string sent by the agent as a Phase and any erroneous status will be visible to the admin.